### PR TITLE
[server] Add and use supervisor image config

### LINKED ISSUE #6512: deploy supervisor image via server ide config
We should move configuration of the supervisor image to the IDE service in Gitpod Server.

### PR DESCRIPTION
**Pull request by corneliusludmann**
## Description

This PR adds the supervisor image to the `server-ide-configmap` and passes this value to the registry facade. That allows to change the supervisor image just by altering the config and thus allows to deploy a new supervisor image by the IDE team on its own.

## Related Issue(s)

Fixes #6512

~~_This aspect is still not solved: https://github.com/gitpod-io/gitpod/issues/6512#issuecomment-963216053. That is why this PR does not close the issue yet._~~

## How to test
- Start a workspace and see in the logs the supervisor version. Log filter:
  ```
  resource.labels.namespace_name="staging-clu-server-supervisor-config"
  jsonPayload.serviceContext.service="supervisor"
  ```
  `jsonPayload.serviceContext.version` has the version.
- Change the supervisor version in [server-ide-config configmap](https://console.cloud.google.com/kubernetes/configmap/europe-west1-b/core-dev/staging-clu-server-supervisor-config/server-ide-config/details?project=gitpod-core-dev), e.g. use version `commit-a7166daa720e8c05c1f200257f567646d427b79b`. No server restart necessary.
- Start another workspace and verify that the version in the logs has been changed.

_**Note:** The version in the configmap can differ from what it is in the logs because images can have multiple tags. E.g. when you have `commit-93104671b75c9f81b2cc8b5a542dc8680e36928f` in the configmap, it's logged as version `commit-20e06321092f0e658622332880889ddda2b77f59` but [you can see in the registry](https://console.cloud.google.com/gcr/images/gitpod-core-dev/eu/build/supervisor?organizationId=585692962993&project=gitpod-core-dev) that both are valid tags for the same image._

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **laushinka** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @laushinka` in a comment when ready.

Associated issue: *#6512*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
@corneliusludmann We haven't really introduced the installer to the teams yet, but it would probably make sense to update [that part](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/ide-configmap.go) accordingly.

**Comment by corneliusludmann** [comment]
> @corneliusludmann We haven't really introduced the installer to the teams yet, but it would probably make sense to update [that part](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/ide-configmap.go) accordingly.

Thanks for the very useful hint! If you don't object, I would like to handle this in a separate PR that also updates other missing changes in the installer: https://github.com/gitpod-io/gitpod/issues/6700

File: chart/templates/registry-facade-configmap.yaml [hunk 1]
```diff
@@ -35,10 +35,6 @@ data:
             "store": "/mnt/cache/registry",
             "requireAuth": false,
             "staticLayer": [
-                {
```
Comment (corneliusludmann) [inline chart/templates/registry-facade-configmap.yaml:38]: @csweichel: Would it be better if we omit this change for now? To be backwards compatible, we need to deploy the new version of `server` before we can remove this configuration from the registry facade, right? If we would deploy this config change before `server` has been updated then we wouldn't have a `supervisor` layer at all.

Thus, it is probably better we keep this as a deprecated state and remove this in 2 weeks or so. WDYT?

Comment (csweichel) [reply chart/templates/registry-facade-configmap.yaml:38]: Good catch. That is indeed correct - let's remove this by the end of the month.

**Comment by JanKoehnlein** [comment]
Hmm, maybe I got it wrong, but it always reports
```
serviceContext: {
service: "supervisor"
version: "commit-20e06321092f0e658622332880889ddda2b77f59"
}
```
here, no matter what I put in as `supervisorImage`

**Comment by corneliusludmann** [comment]
Thanks for carefully testing, @JanKoehnlein! It took me what felt like an eternity of debugging until I figured out what's wrong here. It seems the code is correct but the “How To Test” is a bit tricky.

[According to the logs](https://cloudlogging.app.goo.gl/EAND5cAb419sZhjv7), the server config has been reloaded at the time you tested this PR a couple of times but the supervisor image was always `commit-93104671b75c9f81b2cc8b5a542dc8680e36928f`. Only the last time, the version has been changed to `commit-c55bdc265f2e999994668087283e7900a6daa7e5`. Don't know how many versions you tried. Maybe you were too fast and the server hasn't reloaded the version until you tried the next version. You can see when the server reads a new version with this logs search query:
```
resource.labels.namespace_name="staging-clu-server-supervisor-config"
labels.k8s-pod/component="server"
"ide config updated"
```

And now comes the tricky part. Version `commit-c55bdc265f2e999994668087283e7900a6daa7e5` as well as `commit-93104671b75c9f81b2cc8b5a542dc8680e36928f` of supervisor logs the same version `commit-20e06321092f0e658622332880889ddda2b77f59`. The reason is, that the [build](https://werft.gitpod-dev.com/job/gitpod-build-laushinka-idea-gitignore.0) that has been built the new version `commit-c55bdc265f2e999994668087283e7900a6daa7e5` has changes in the supervisor Dockerfile only and the Docker build used the cached version of the supervisor app. That's why the 2 different supervisor Docker images use the same supervisor app binary with the same version. However, when you use a version like `main.1830` or `commit-a7166daa720e8c05c1f200257f567646d427b79b` you get actually a different version.

**TL;DR:**

Using a different Docker version tag does not necessarily mean that you get another supervisor app binary with a different logged version. However, use these 2 tags and you should get different versions logged:
```
commit-20e06321092f0e658622332880889ddda2b77f59
commit-a7166daa720e8c05c1f200257f567646d427b79b
```

**Comment by akosyakov** [comment]
@gitpod-io/engineering-meta Could someone finish the review please? 🙏

**Review by geropl — APPROVED** [review]
Cannot test myself bc of cert issues but:
 - I see others have tested before
 - I understand the mechanisms at play thx to @corneliusludmann nice explanation
 - code LGTM
 :+1:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 29f1dcf44fb9215a062737f59983525ac37cb839</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6669#pullrequestreview-809638674" title="Approved">geropl</a>*

Associated issue: *#6512*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
